src/authentication.js

```diff
@@ -154,8 +154,14 @@ export class Authentication {
       }
     }
 
-    this.payload = payload;
-    this.exp = payload ? parseInt(payload.exp, 10) : NaN;
+    this.payload = payload || response;
+
+    this.exp = NaN;
+    if (this.payload) {
+      const exp = this.payload[this.config.accessTokenExpProp];
```

Comment:
I've never used NaN as a default value personally, but I suppose it can be fine. I don't really dig the const though.